src/defense.cpp

```diff
@@ -189,9 +189,9 @@ void defense_game::init_mtypes()
 
 void defense_game::init_constructions()
 {
-    for( auto &c : constructions ) {
-        c->time = 1; // Everything takes 1 minute
-    }
+    for_each_construction([&](construction& c) {
```

Comment:
Instead of exposing a for_each wrapper that is only called once, just expose a method that sets all the times to 1.
Keep interfaces as small and simple as possible to avoid unnecessary coupling.